File: src/test/java/com/emc/mongoose/cli/CliArgUtilTest.java
package com.emc.mongoose.cli;

import org.junit.Test;
import static org.junit.Assert.assertEquals;

import java.util.Map;

public class CliArgUtilTest {

	@Test
	public final void test()
	throws Exception {
		final String[] args = new String[] {
			"--name=goose",
			"--io-buffer-size=1KB-4MB",
			"--storage-node-http-headers=customHeaderName:customHeaderValue",
			"--enable-some-feature",
		};
		final Map<String, String> parsedArgs = CliArgUtil.parseArgs(args);
		assertEquals("goose", parsedArgs.get("name"));
		assertEquals("1KB-4MB", parsedArgs.get("io-buffer-size"));
		assertEquals("customHeaderName:customHeaderValue", parsedArgs.get("storage-node-http-headers"));
		assertEquals(Boolean.TRUE.toString(), parsedArgs.get("enable-some-feature"));
	}

}

File: src/test/scala/com/emc/mongoose/cli/CliArgUtilTest.scala
package com.emc.mongoose.cli

import com.emc.mongoose.config.CliArgUtil

import org.junit.Assert.assertEquals
import org.junit.Test

import java.lang.Boolean.TRUE

final class CliArgUtilTest {

	@Test @throws[Exception]
	def test(): Unit = {
		val parsedArgs = CliArgUtil parseArgs(
			"--name=goose",
			"--io-buffer-size=1KB-4MB",
			"--storage-node-http-headers=key0:val0",
			"--storage-node-http-headers=key1:val1",
			"--enable-some-feature",
		)
		assertEquals("goose", parsedArgs get "name")
		assertEquals("1KB-4MB", parsedArgs get "io-buffer-size")
		val httpHeaders = new java.util.HashMap[String, String]()
		httpHeaders put("key0", "val0")
		httpHeaders put("key1", "val1")
		assertEquals(httpHeaders, parsedArgs get "storage-node-http-headers")
		assertEquals(TRUE toString, parsedArgs get "enable-some-feature")
	}
}
